qiskit_dynamics/solver_classes.py

```diff
@@ -0,0 +1,335 @@
+            drift: Hamiltonian drift operator.
+            rotating_frame: Rotating frame to transform the model into.
+            evaluation_mode: Method for model evaluation. See documentation for
+                             :meth:`HamiltonianModel.set_evaluation_mode` or
+                             :meth:`LindbladModel.set_evaluation_mode`
+                             (if dissipators in model) for valid modes.
+            rwa_cutoff_freq: Rotating wave approximation cutoff frequency. If ``None``, no
+                             approximation is made.
+        """
+
+        model = None
+        if dissipator_operators is None:
+            model = HamiltonianModel(
+                operators=hamiltonian_operators,
+                signals=hamiltonian_signals,
+                rotating_frame=rotating_frame,
+                drift=drift,
+                evaluation_mode=evaluation_mode,
+            )
+            self._signals = hamiltonian_signals
+        else:
+            model = LindbladModel(
+                hamiltonian_operators=hamiltonian_operators,
+                hamiltonian_signals=hamiltonian_signals,
+                dissipator_operators=dissipator_operators,
+                dissipator_signals=dissipator_signals,
+                rotating_frame=rotating_frame,
+                drift=drift,
+                evaluation_mode=evaluation_mode,
+            )
+            self._signals = (hamiltonian_signals, dissipator_signals)
+
+        self._rwa_signal_map = None
+        if rwa_cutoff_freq is not None:
+            model, rwa_signal_map = rotating_wave_approximation(
+                model, rwa_cutoff_freq, return_signal_map=True
+            )
+            self._rwa_signal_map = rwa_signal_map
+
+        self._model = model
+
+    @property
+    def rotating_frame(self) -> RotatingFrame:
+        """Rotating frame of the model."""
+        return self.model.rotating_frame
+
+    @rotating_frame.setter
+    def rotating_frame(self, new_rotating_frame: Union[Array, RotatingFrame]):
+        """Set rotating frame for the model."""
+        self.model.rotating_frame = new_rotating_frame
+
+    @property
+    def model(self) -> BaseGeneratorModel:
+        """The model."""
+        return self._model
+
+    @property
+    def signals(self) -> SignalList:
+        """Solver signals."""
+        return self._signals
+
+    @signals.setter
+    def signals(self, new_signals: Union[List[Signal], SignalList]):
+        """Set signals for the solver, and pass to the model."""
+        self._signals = new_signals
+        if self._rwa_signal_map is not None:
+            new_signals = self._rwa_signal_map(new_signals)
+        self.model.signals = new_signals
+
+    @property
+    def evaluation_mode(self) -> str:
+        """Evaluation mode for the model."""
+        return self.model.evaluation_mode
+
+    def set_evaluation_mode(self, new_mode: str):
```

Comment:
We'll get rid of this one here anyway, but in the model classes this one was changed to a method so that users can call `help(...)` on it to see what the available methods are for each model. 

In general I have been finding the usage of `setters` a bit awkward and keep going back and forth on whether to get rid of them and just use methods with `.set_...`. It doesn't seem like too huge a deal as deprecating/shifting doesn't seem like too big a deal in this case if we do decide to change later.